Entities/Common/Factory/Shop.as

```diff
@@ -91,201 +92,214 @@ void onCommand(CBlob@ this, u8 cmd, CBitStream @params)
 	}
 	else if (cmd == this.getCommandID("shop buy"))
 	{
-		if (this.hasTag("shop disabled"))
-			return;
+		BuyShopItem(this, params);
+	}
+	else if (cmd == this.getCommandID("shop buy hotkey"))
+	{
+		CControls@ controls = getControls();
```

Comment:
I don't think this will work on server - controls object either wont exist or at least wont be the same as the player's (how could it be, for every player). You'll need to write the command to buy clientside and find some other way to sanity-check serverside.

Reply:
@eps0003 any thoughts on this? gonna need a workaround before it can be merged unfortunately